modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -178,8 +203,21 @@ private void _validate(ObjectEntry objectEntry) {
 						"r_apiApplicationToAPIEndpoints_c_apiApplicationId"))) {
 
 				throw new ObjectEntryValuesException.InvalidObjectField(
-					"An API endpoint must be related to a an API application",
-					"an-api-endpoint-must-be-related-to-a-an-api-application",
+					"An API endpoint must be related to an API application",
+					"an-api-endpoint-must-be-related-to-an-api-application",
+					null);
+			}
+
+			if (!_isAPISchema(
+					(long)values.get(
+						"r_requestAPISchemaToAPIEndpoints_c_apiSchemaId")) ||
```

Comment:
Regarding on the questions above, it shouldn't be false because you will assume that an endpoint should have a request schema and a reponse schema. In order to make it more readable, we can extract the `if (apiApplicationId == 0) {` from the method